pkg/render/egressgateway/egressgateway.go

```diff
@@ -0,0 +1,438 @@
+// Copyright (c) 2023 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package egressgateway
+
+import (
+	"fmt"
+	"strings"
+
+	ocsv1 "github.com/openshift/api/security/v1"
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/ptr"
+	"github.com/tigera/operator/pkg/render"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/podsecuritypolicy"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	egwPortName               = "health"
+	DefaultEGWVxlanPort int   = 4790
+	DefaultEGWVxlanVNI  int   = 4097
+	DefaultHealthPort   int32 = 8080
+)
+
+func EgressGateway(
+	config *Config,
+) render.Component {
+	return &component{
+		config: config,
+	}
+}
+
+type component struct {
+	config *Config
+}
+
+// Config contains all the config information EgressGateway needs to render component.
+type Config struct {
+	PullSecrets  []*corev1.Secret
+	Installation *operatorv1.InstallationSpec
+	OSType       rmeta.OSType
+	EgressGW     *operatorv1.EgressGateway
+
+	egwImage          string
+	EgressGWVxlanVNI  int
```

Comment:
nit: VXLAN instead of Vxlan

Reply:
Or even just "VXLANVNI" and "VXLANPort" since we don't need to re-write "EgressGW" - the context is already clear.